src/PostgREST/Query/QueryBuilder.hs

```diff
@@ -209,6 +218,23 @@ callPlanToQuery (FunctionCall qi params args returnsScalar returnsSetOfScalar re
     fmtParams prms = intercalateSnippet ", "
       ((\a -> (if ppVar a then "VARIADIC " else mempty) <> pgFmtIdent (ppName a) <> " := pgrst_body." <> pgFmtIdent (ppName a)) <$> prms)
 
+    fmtArgs :: [RoutineParam] -> HM.HashMap Text RpcParamValue -> SQL.Snippet
+    fmtArgs prms args = intercalateSnippet ", " $ fmtArg <$> prms
+      where
+        fmtArg RoutineParam{..} =
+          (if ppVar then "VARIADIC " else mempty) <>
+          pgFmtIdent ppName <>
+          " := " <>
+          encodeArg (HM.lookup ppName args) <>
+          "::" <>
+          SQL.sql (encodeUtf8 ppTypeMaxLength)
+        encodeArg :: Maybe RpcParamValue -> SQL.Snippet
+        encodeArg (Just (Variadic v)) = SQL.encoderAndParam (HE.nonNullable $ HE.foldableArray $ HE.nonNullable HE.text) v
+        encodeArg (Just (Fixed v)) = SQL.encoderAndParam (HE.nonNullable HE.unknown) $ encodeUtf8 v
+        -- Currently not supported: Calling functions without some of their arguments without DEFAULT.
+        -- We could fallback to providing this NULL value in those cases.
+        encodeArg Nothing = "NULL"
```

Comment:
How about a test for this case? Would help in understanding and proving the limitation

Reply:
> But you could argue that if there is no exact match, then we can just fallback to partial matches and call rpc1(1, NULL) in this case

Oh, this might be a problem for typing and OpenAPI. cc @laurenceisla 

Also not sure if we want that behavior. In postgres, doing:

```sql
-- postgrest-with-postgresql-15  -f test/spec/fixtures/load.sql psql -U postgres

select add_them(a:=3);
ERROR:  function add_them(a => integer) does not exist
LINE 1: select add_them(a:=3);
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.

-- select add_them(a:=3, b:=9); succeeds
```

Results in an error. The query string should be equivalent to named arguments, I believe.